### PR TITLE
expression: Add cast as json push down support for tiflash

### LINKED ISSUE #48551: Add cast as json push down support for tiflash
## Enhancement

https://github.com/pingcap/tiflash/issues/8307

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: close #48551

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by SeaRise** [comment]
/hold
wait for https://github.com/pingcap/tiflash/pull/8333

**Review by XuHuaiyu — APPROVED** [review]

File: pkg/expression/expr_to_pb_test.go [hunk 1]
```diff
@@ -1294,6 +1294,44 @@ func TestExprPushDownToFlash(t *testing.T) {
...
+	function, err = NewFunction(mock.NewContext(), ast.Cast, types.NewFieldType(mysql.TypeJSON), datetimeColumn)
+	require.NoError(t, err)
+	exprs = append(exprs, function)
+
+	// CastDurationAsJson
+	function, err = NewFunction(mock.NewContext(), ast.Cast, types.NewFieldType(mysql.TypeJSON), durationColumn)
+	require.NoError(t, err)
+	exprs = append(exprs, function)
+
+	// CastDurationAsJson
```
Comment (windtalker) [inline pkg/expression/expr_to_pb_test.go:1330]:
```suggestion
	// CastJsonAsJson
```

**Review by windtalker — APPROVED** [review]
Others LGTM

**Pull request merged**